integration_tests/test_suites/k8s-test-suite/tests/test_external_asset.py

```diff
@@ -164,3 +166,50 @@ def number_y(
     mats = result.asset_materializations_for_node(number_y.op.name)
     assert "is_even" in mats[0].metadata
     assert mats[0].metadata["is_even"].value is True
+
+
+def test_use_excute_k8s_job(namespace, cluster_provider):
+    docker_image = get_test_project_docker_image()
+
+    @asset
+    def number_y_job(context: AssetExecutionContext):
+        core_api = kubernetes.client.CoreV1Api()
+        reader = K8sPodLogsMessageReader()
+        with setup_ext_protocol(
+            context,
+            ExtEnvContextInjector(),
+            reader,
+            extras={"storage_root": "/tmp/"},
+        ) as protocol_env:
+            job_name = "number_y_asset_job"
+
+            # stand-in for any means of creating kubernetes work
+            execute_k8s_job(
+                context=context,
+                image=docker_image,
+                command=[
+                    "python",
+                    "-m",
+                    "numbers_example.number_y",
+                ],
+                env_vars=[
+                    f"{k}={v}"
+                    for k, v in {
+                        "PYTHONPATH": "/dagster_test/toys/external_execution/",
+                        "NUMBER_Y": "2",
+                        **protocol_env,
+                    }.items()
+                ],
+                k8s_job_name=job_name,
+            )
```

Comment:
could replace this block with for example, running some existing script that produces a Kubeflow CRD  😉